src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -646,16 +662,30 @@ public void doStatus(@QueryParameter String token) {
             System.err.println("Checking status");
             checkToken(token);
         }
+        /**
+         * Used to run test methods on a separate thread so that code that uses {@link Stapler.getCurrentRequest}
+         * does not inadvertently interact with the request for {@link #doStep} itself.
```

Comment:
I would be fine reverting this part of the PR if anyone has any concerns that are not easy to address.